source/core/CEntryPoints.cpp

```diff
@@ -210,6 +210,29 @@ VIREO_EXPORT Int32 EggShell_GetArrayDimLength(TypeManagerRef tm, const char* viN
     return actualArray->GetLength(dim);
 }
 //------------------------------------------------------------
+//! Resizes a variable size Array symbol to have new dimension lengths specified by newLengths, it also initializes cells for non-flat data.
+//! Retuns -1 if the symbols is not found, -2 if was not possible to resize the array and 0 if resizing was successful.
+VIREO_EXPORT Int32 EggShell_ResizeArray(TypeManagerRef tm, const char* viName, const char* eltName, Int32 rank, Int32* newLengths)
+{
+    TypeManagerScope scope(tm);
+
+    SubString objectName(viName);
+    SubString path(eltName);
+    void *pData = null;
+
+    TypeRef actualType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
+    if (actualType == null || !actualType->IsArray()) {
+        return -1;
```

Comment:
If you are going to return more than one error code, make them positive and use standard LabVIEW-style error codes.
kLVError_ArgError (=1)  for bad type/null array, and kLVError_MemFull (=2) for resize failure.
Add these errors to enum LVError in DataTypes.h.

I'll make a pass and modify the other prims which return local error codes which are not actually prim-specific to use these too.